pymeasure/instruments/agilent/agilentE5062A.py

```diff
@@ -45,6 +45,362 @@
+        validator=strict_discrete_set,
+        values=['S11', 'S12', 'S21', 'S22']
+    )
+
+
+class VNAChannel(Channel):
+    """A measurement channel of the E5061A/E5062A"""
+
+    def _add_trace(self, override_id=None):
+        """Internal method that adds a trace to the traces list"""
+        tr_id = len(self.traces) + 1 if override_id is None else override_id
+        self.add_child(VNATrace, tr_id, 'traces', 'tr_')
+
+    def _update_trace_count(self, active_traces):
+        """Internal method that updates the trace list based on the number
+        of active traces.
+        """
+        while len(self.traces) > active_traces:
+            self.remove_child(self.traces[max(self.traces.keys())])
+        while len(self.traces) < active_traces:
+            self._add_trace()
+
+    def _update_power_values(self, attenuation):
+        """Internal method that updates the valid range of stimulus power
+        based on the configured attenuation"""
+        self.power_values = (
+            -attenuation - 5,
+            -attenuation + 10)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # self.traces = []
+        self._add_trace(override_id=1)                # add just a single trace to init it all
+        self._update_trace_count(self.active_traces)  # add remaining traces if necessary
+        self._update_power_values(self.attenuation)   # update dynamic limits of power
+
+    start_frequency = Channel.control(
+        "SENSe{ch}:FREQuency:STARt?", "SENSe{ch}:FREQuency:STARt %g",
+        """Control the start frequency in Hz (float).""",
+        cast=lambda x: int(float(x)),
+        validator=strict_range,
+        values=[3e5, 3e9]
+    )
+
+    stop_frequency = Channel.control(
+        "SENSe{ch}:FREQuency:STOP?", "SENSe{ch}:FREQuency:STOP %g",
+        """Control the stop frequency in Hz (float).""",
+        cast=lambda x: int(float(x)),
+        validator=strict_range,
+        values=[3e5, 3e9]
+    )
+
+    scan_points = Channel.control(
+        "SENSe{ch}:SWEep:POINts?", "SENSe{ch}:SWEep:POINts %g",
+        """Control the number of points used in a sweep (int). Valid range 2 - 1601.""",
+        cast=lambda x: int(float(x)),
+        validator=strict_range,
+        values=[2, 1601]
+    )
+
+    sweep_time = Channel.control(
+        "SENSe{ch}:SWEep:TIME?", "SENSe{ch}:SWEep:TIME %g",
+        """Control the sweep time in seconds (float). The allowable range
+        varies on config and the set value is truncated. Note that
+        ``sweep_time_auto`` needs to be ``False`` for changes to this
+        property to have an effect."""
+    )
+
+    sweep_time_auto = Channel.control(
+        "SENSe{ch}:SWEep:TIME:AUTO?", "SENSe{ch}:SWEep:TIME:AUTO %d",
+        """Control whether to automatically set the sweep time (bool). You
+        probably want this on to always keep the sweep time to a minimum (given
+        the range, IF BW, and sweep delay time).""",
```

Comment:
For boolean properties names like `sweep_time_auto_enabled` are best, as they indicate directly, what True/False mean.